src/components/bmfont-text.js

```
@@ -0,0 +1,333 @@
+        var shaderObject = this.shaderObject = new ShaderType();
+        shaderObject.el = this.el;
+        shaderObject.init(data);
+        shaderObject.update(data);
+        shader = shaderObject.material;
+        shader.transparent = data.transparent; // apparently this is not set on either init or update
+        this.material = shader;
+        if (this.mesh) { this.mesh.material = this.material; }
+        return;
+      }
+      if (this.data.shader === 'SDF') {
+        shader = createSDF(data);
+      } else if (this.data.shader === 'MSDF') {
+        shader = createMSDF(data);
+      } else {
+        shader = createBasic(data);
+      }
+      this.material = new THREE.RawShaderMaterial(shader);
+    } else if (this.data.shader === 'custom') {
+      this.shaderObject.update(data);
+      this.shaderObject.material.transparent = data.transparent; // apparently this is not set on either init or update
+    } else {
+      this.material.uniforms.opacity.value = this.data.opacity;
+      this.material.uniforms.color.value.set(this.data.color);
+      this.material.uniforms.map.value = this.texture;
+    }
+
+    if (this.mesh) { this.mesh.material = this.material; }
+  },
+
+  registerFont: function (key, url) { fontMap[key] = url; },
+
+  lookupFont: function (keyOrUrl) { return fontMap[keyOrUrl] || keyOrUrl; },
+
+  loadBMFontPromise: function (src) {
+    return new Promise(function (resolve, reject) {
+      loadBMFont(src, function (err, font) {
+        if (err) { reject(err); } else { resolve(font); }
+      });
+    });
+  },
+
+  loadTexturePromise: function (src) {
+    return new Promise(function (resolve, reject) {
+      var loader = new THREE.ImageLoader();
+      loader.load(src, function (image) {
+        resolve(image);
+      }, undefined, function () {
+        reject(null);
+      });
+    });
+  },
+
+  updateFont: function () {
+    if (!this.data.font) {
+      console.error(new TypeError('No font specified for bmfont text!'));
```

Do you want to throw this error?